agent/ecscni/types_linux.go

```diff
@@ -159,3 +164,39 @@ type BranchENIConfig struct {
 	// InterfaceType is the type of the interface to connect the branch ENI to
 	InterfaceType string `json:"interfaceType,omitempty"`
 }
+
+type ServiceConnectConfig struct {
```

Comment:
I believe this is the recommended way actually, according to https://go.dev/doc/effective_go#package-names
> The importer of a package will use the name to refer to its contents, so exported names in the package can use that fact to avoid repetition

In our case, the package name `ecscni` already implies the namespace, and other packages will be referencing the type as `ecscni.ServiceConnectConfig`.